rediscluster/connection.py

```diff
@@ -220,9 +220,10 @@ def make_connection(self, node):
         """
         Create a new connection
         """
-        if self.count_all_num_connections(node) >= self.max_connections:
+        num_connections = self.count_all_num_connections(node)
```

Comment:
This is a subtle change, but just prevents `self.count_all_num_connections` from needing to be called and calculated twice if/when the exception is raise.